testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergAzure.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.iceberg;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import io.trino.tempto.BeforeMethodWithContext;
+import io.trino.tempto.ProductTest;
+import org.testng.annotations.Test;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tests.product.TestGroups.ICEBERG_AZURE;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestIcebergAzure
+        extends ProductTest
+{
+    @Inject
+    @Named("databases.presto.abfs_schema")
+    private String schema;
+    private String schemaLocation;
+
+    @BeforeMethodWithContext
+    public void setUp()
+    {
+        String container = requireNonNull(System.getenv("ABFS_CONTAINER"), "Environment variable not set: ABFS_CONTAINER");
+        String account = requireNonNull(System.getenv("ABFS_ACCOUNT"), "Environment variable not set: ABFS_ACCOUNT");
+        schemaLocation = format("abfs://%s@%s.dfs.core.windows.net/%s", container, account, schema);
+    }
+
+    @Test(groups = {ICEBERG_AZURE, PROFILE_SPECIFIC_TESTS})
+    public void testCreateAndSelectNationTable()
+    {
+        String tableName = "nation_" + randomNameSuffix();
+        String tableLocation = schemaLocation + "/" + tableName;
+        onTrino().executeQuery(format(
+                "CREATE TABLE iceberg.default.%1$s WITH (location = '%2$s/%1$s') AS SELECT * FROM tpch.tiny.nation",
+                tableName,
+                tableLocation));
+
+        assertThat(onTrino().executeQuery("SELECT count(*) FROM iceberg.default." + tableName)).containsOnly(row(25));
+        onTrino().executeQuery("DROP TABLE iceberg.default." + tableName);
+    }
+
```

Comment:
I'd like to see some corner test case scenarios : special characters, double slashes.

Reply:
You mean similar to TestHiveS3AndGlueMetastoreTest for gcs and azure, right? not necessarily product-tests?

Reply:
[TestAzureBlobFileSystem](https://github.com/trinodb/trino/blob/master/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestAzureBlobFileSystem.java) already exist for Azure. But for GCS there isnt. The product-tests are spread all over the place :( I will try to organize this a bit